Version 2?

Alexandre-T opened the pull request:
Hi,

I needed a new version of doctrine2-spatial which is compatible with PHP 7.2+. So I forked your excellent library and updated it. Because of a lot of changes between PHP 5.1 and PHP 7, my updates do **NOT** warranty backward compatibility. So, if you're agree to merge my pull-request in your repository, you're welcomed, but do not forget that you should create a new major version.

All new features are described in the [Changelog file](https://github.com/Alexandre-T/doctrine2-spatial/blob/master/CHANGELOG.md). There is a [new documentation](https://doctrine2-spatial.readthedocs.io/en/latest/). _Please, forgive me, I'm not fluent in english._

If you merge it, I will update the documentation to replace references to my package `alexandret/doctrine2-spatial` by yours.

You can find the travis, code climat and code coverage results here : 

[![Build Status](https://travis-ci.org/Alexandre-T/doctrine2-spatial.svg?branch=master)](https://travis-ci.org/Alexandre-T/doctrine2-spatial)
[![Code Climate](https://codeclimate.com/github/Alexandre-T/doctrine2-spatial/badges/gpa.svg)](https://codeclimate.com/github/Alexandre-T/doctrine2-spatial)
[![Coverage Status](https://coveralls.io/repos/Alexandre-T/doctrine2-spatial/badge.svg?branch=master&service=github)](https://coveralls.io/github/Alexandre-T/doctrine2-spatial?branch=master)
[![Documentation Status](https://readthedocs.org/projects/doctrine2-spatial/badge/?version=latest)](https://doctrine2-spatial.readthedocs.io/en/latest/?badge=latest)

Alexandre

Alexandre-T committed:
Test type of value before using ctype function

Alexandre-T committed:
Test type of value before using ctype function

Alexandre-T committed:
Fork and code quality and code test updated

Alexandre-T committed:
Void added to setUp method, Php-cs-fixer fix

Alexandre-T committed:
Replacing CoversNothing by CoversDefaultClass

Alexandre-T committed:
Fixing code sniffer errors and warnings

Alexandre-T committed:
Fixing code sniffer errors and warnings

Alexandre-T committed:
Fixing code sniffer errors and warnings

Alexandre-T committed:
Readme updated

Alexandre-T committed:
Code climat test coverage

Alexandre-T committed:
Installing postgresql server and postgis extension

Alexandre-T committed:
Phpunit configuration updated

Alexandre-T committed:
Comments improved

Alexandre-T committed:
Converts methods With PHP to camel case (Php)

Alexandre-T committed:
Fixing syntax

Alexandre-T committed:
Only stable upcoming releases are tested

Alexandre-T committed:
Only stable upcoming releases are tested

Alexandre-T committed:
Fixing phpunit.xml.dist

Alexandre-T committed:
Try postgis2.5

Alexandre-T committed:
Try postgis2.5

Alexandre-T committed:
Trying mysql

Alexandre-T committed:
Fixing composer to test doctrine 3.0 version

Alexandre-T committed:
Converting methods to camel case

Alexandre-T committed:
Comments improved

Alexandre-T committed:
Test simplified

Alexandre-T committed:
Methods move into parent class

Alexandre-T committed:
Methods move into parent class

Alexandre-T committed:
Documentation for tests improved

Alexandre-T committed:
Documentation for tests improved

Alexandre-T committed:
Test code simplified

Alexandre-T committed:
Helper created to simplify tests

Alexandre-T committed:
3 Helpers created to simplify tests

Alexandre-T committed:
phpcsfixer

Alexandre-T committed:
Upgrade of tests finished
preparing php mess detector

Alexandre-T committed:
Upgrade of tests finished
preparing php mess detector

Alexandre-T committed:
Updating travis to test doctrine 3 only with php 7.4

Alexandre-T committed:
Adding return types

Alexandre-T committed:
Quality should respect quality code rules to be merged

Alexandre-T committed:
Clodeclimate updated to avoid similar code process
because it is not revelent with structure of this library

Alexandre-T committed:
Abstract Geometry and all its children updated with methods insteadof properties.
Method help test coverage to detect non-tested function (and there are a lot!)

Alexandre-T committed:
Composer quality wont break a pull request

Alexandre-T committed:
license with http link replaced by https link.

Alexandre-T committed:
http link replaced by https link.

Alexandre-T committed:
Updated

Alexandre-T committed:
PHPUnit method replaced by PHPUnit static method

Alexandre-T committed:
Documentation updated

Alexandre-T committed:
Splitting tests and MySQL versions

Alexandre-T committed:
Fixing typo in travis

Alexandre-T committed:
Before_script was not executed

Alexandre-T committed:
Before_script was not executed

Alexandre-T committed:
Before_script was not executed

Alexandre-T committed:
Before_script was not executed

Alexandre-T committed:
Fixing link

Alexandre-T committed:
Spatial functions respecting OGC Standard implemented

Alexandre-T committed:
Spatial functions respecting OGC Standard implemented

Alexandre-T committed:
Updating Travis and installing postgis on postgresql10

Alexandre-T committed:
Updating postgis on postgresql10 and 11

Alexandre-T committed:
Removing docker for postgresql 11

Alexandre-T committed:
Removing docker for postgresql 11

Alexandre-T committed:
Manual installation of postgresql 11, automatic installation of postgresql 10

Alexandre-T committed:
StDistanceTest.php renamed

Alexandre-T committed:
St_Intersection

Alexandre-T committed:
St_Difference

Alexandre-T committed:
St_Union implemented and tested

Alexandre-T committed:
St_SymDifference implemented and tested

Alexandre-T committed:
ST_Buffer implemented and tested
Mysql Variant implemented too

Alexandre-T committed:
ST_ConvexHull implemented but not tested

Alexandre-T committed:
Travis updated

Alexandre-T committed:
ST_X and ST_Y done and tested

Alexandre-T committed:
ST_StartPoint and ST_EndPoint done and tested

Alexandre-T committed:
ST_IsRing and ST_Length done and tested

Alexandre-T committed:
Fixing some bug and implementing ScGeographyFromText

Alexandre-T committed:
Trying to fix connection on PostGreSQL

Alexandre-T committed:
Trying to fix connection on PostGreSQL

Alexandre-T committed:
Update README.md

Alexandre-T committed:
Fixing uppercase in directory (windows powa)

Alexandre-T committed:
Update README.md

Alexandre-T committed:
Fixing test and Travis execution plan

Alexandre-T committed:
Fixing a misconfiguration

Alexandre-T committed:
Adding a TODO

Alexandre-T committed:
Adding a TODO

Alexandre-T committed:
Adding spatial functions and their tests

Alexandre-T committed:
Adding MultiPointType, spatial functions and their tests

Alexandre-T committed:
Replacing classname

Alexandre-T committed:
Testing coversDefaultClass

Alexandre-T committed:
Fixing typo in column name

Alexandre-T committed:
ST_IsClosed and unspecified GeometryType functions

Alexandre-T committed:
PgSql specific functions

Alexandre-T committed:
PgSql specific functions

Alexandre-T committed:
PgSql specific functions (tests missing)

Alexandre-T committed:
PgSql specific tests

Alexandre-T committed:
PgSql specific tests

Alexandre-T committed:
PgSql specific functions

Alexandre-T committed:
Removing unused scripts

Alexandre-T committed:
Typo fixed

Alexandre-T committed:
Fixing travis which can failed because of cached files

Alexandre-T committed:
Fixing test

Alexandre-T committed:
SP_Scale and SP_Split PostgreSQL functions

Alexandre-T committed:
SP_Simplify and SP_NPoints PostgreSQL functions

Alexandre-T committed:
SP_SnapToGrid and SP_DistanceSphere PostgreSQL functions

Alexandre-T committed:
Four new functions

Alexandre-T committed:
Trying to fix SqlError on linux

Alexandre-T committed:
Adding new PostgreSQL functions

Alexandre-T committed:
Adding new PostgreSQL test

Alexandre-T committed:
Adding new PostgreSQL functions and tests

Alexandre-T committed:
Fixing small issues

Alexandre-T committed:
Two new PostgreSql function

Alexandre-T committed:
New geometry constructors

Alexandre-T committed:
MySql MBR functions

Alexandre-T committed:
Removing deprecated functions

Alexandre-T committed:
PhpCsFixer

Alexandre-T committed:
Php code sniffer

Alexandre-T committed:
Php mess detector rules updated

Alexandre-T committed:
Trying to fix codecoverage

Alexandre-T committed:
Fixing missing classes in codecoverage

Alexandre-T committed:
Integration test on doctrine 2.8.x-dev added

Alexandre-T committed:
Documentation updated

Alexandre-T committed:
Stages updated

Alexandre-T committed:
Merge branch 'ogc'

Alexandre-T committed:
documentation

Alexandre-T committed:
Initialization of sphinx documentation

Alexandre-T committed:
Initialization of template

Alexandre-T committed:
First documentation pages

Alexandre-T committed:
Small update for main file

Alexandre-T committed:
Small update for main file

Alexandre-T committed:
Files moved to be compliant with readthedocs.io

Alexandre-T committed:
Files moved to be compliant with readthedocs.io

Alexandre-T committed:
Documentation structure done

Alexandre-T committed:
Documentation improved

Alexandre-T committed:
Documentation improved

Alexandre-T committed:
Documentation improved

Alexandre-T committed:
Documentation done!

Alexandre-T committed:
Fixing documentation badge and codeclimate configuration

Alexandre-T committed:
Geometry and Geography implements JsonSerializable
https://github.com/creof/doctrine2-spatial/pull/194

Alexandre-T committed:
Version 2.0.0

Alexandre-T committed:
Update README to fix links with initial repository

sadortun commented:
Wow, awesome work !

djlambert commented:
@sadortun agreed!

I haven't had time to contribute for quite a while, and haven't been working with PHP much the last couple years. @sadortun maybe you're in a better position to merge this? It looks like coveralls is being picky. I have 4k monitors and could care less about line lengths.

cedvan commented:
:+1:

cedvan commented:
Hi,

I find a bug, postgresql function `ST_Distance_Sphere` was renamed in postgresql by `ST_DistanceSphere`. 
So fix is necessary in `CrEOF\Spatial\ORM\Query\AST\Functions\PostgreSql\SpDistanceSphere`

See you

Alexandre-T commented:
I will edit it. But, this bug is surprising, because this function is tested in this [functional test](https://github.com/Alexandre-T/doctrine2-spatial/blob/master/tests/CrEOF/Spatial/Tests/ORM/Query/AST/Functions/PostgreSql/SpDistanceSphereTest.php). My test should failed and it didn't! Why?

edit: Oh, ok! The function was renamed on postgis!

Alexandre-T commented:
@sadortun , do you think you can merge this as suggested by @djlambert ? If you can't, how can I help you?

sadortun commented:
@Alexandre-T can you fix the merge conflicts ?


Thanks

tugrul commented:
Please don't change `protected` fields to `private`. This could be break existing codebase of projects. `protected` is better to extend library code without touch library's source. `private` is logical for rare cases like keep plaintext password before hash because we don't want leak the information possible flaw or if some classes has many execution steps and if there a possible break when we touch a member of the class then we can declare as `private`. But we should not use always `private` because it kills flexibility.

Alexandre-T committed:
Fixing merge conflicts (1/3)
private property replace into a protected property.
removing final key words.
Removing references to alexandret package for merge in creof project.

Alexandre-T committed:
Fixing a bug on Sp_DistanceSphere

Alexandre-T committed:
Fixing a link on doctrine persistance 2.0.0
(no incidence in code because class is only used in comments)

Alexandre-T committed:
Fixing merge conflicts on documentation

Alexandre-T committed:
Forcing PhpUnit8
On phpunit9 assertRegExp is deprecated.
I can't use phpunit9, which is not compatible with php7.2

Alexandre-T committed:
Documentation updated

Alexandre-T committed:
PhpCsFixer

Alexandre-T committed:
Testing postgis 2.4 with postgresql 10

Alexandre-T committed:
Fixing travis for doctrine/orm 2.8.dev

Alexandre-T committed:
Fixing travis

Alexandre-T committed:
Fixing test

Alexandre-T committed:
Email fixed
File test renamed

Alexandre-T committed:
Fix Code Climate config and MySql unused parameter

Alexandre-T committed:
Code complexity removed

Alexandre-T committed:
Code complexity removed

Alexandre-T commented:
@tugrul this [commit](https://github.com/creof/doctrine2-spatial/pull/209/commits/628894c20ea66eae01633712e0cb6999252ba9b7) removes `final` keywords and updates private properties to protected.
@cedvan this [commit](https://github.com/creof/doctrine2-spatial/pull/209/commits/f5cf0756a9845666da96ec7f4b878b9ab2f13c3d) replace the old postgis function name by the new one (ST_DistanceSphere). I created another class for the deprecated function (ST_Distance_Sphere)
@sadortun I've removed references to my fork `alexandret/doctrine2-spatial` as you can see in [this version of the documentation](https://doctrine2-spatial.readthedocs.io/en/merge-creof/). When this version will be merged, I will update URL of the repository in the readthedocs.org website. I cannot do it actually, I think it's because the creof/doctrine2-spatial repository has no directory containing documentation source, yet. 

This [travis build](https://travis-ci.org/github/Alexandre-T/doctrine2-spatial/builds/695178976) is corresponding to the last commit. As you can see, all build jobs are "green".

florida63 commented:
@Alexandre-T 
Thanks for this PR

Alexandre-T commented:
@sadortun Is there anything else blocking the merge?

sadortun commented:
@djlambert  @Alexandre-T I really appreciate the time and effort you put in, and we will merge this soon!.


@djlambert Just to make sure we dont cause a major panic out there,  I want your opnion on this.

1)
Since this PR **might** include BC breaks. it should be bumped to a next major version tag, that prety obvious. But i can see an issue where folks are using `dev-master` will get BC break. Should we still merge this into `master` ?

Here the steps i propose :
- Tag current master into `1.3.0`
- Create branch `1.x` based on current `master`
- Merge this PR.
- Tag the updated `master` into `2.0,0`

2)
Will this cause issues with other @creof libraries ?

3)
We probably need a quick note in `Readme` to explain that "If you are using `dev-master` and got BC issue after a `composer update` that you should tag your dependency on `~1.2`, or follow  some instructions to help to get through the update.

4) 
@Alexandre-T  We probably need, if not already done, update the `CHANGELOG` in this PR

tugrul commented on lib/CrEOF/Spatial/PHP/Types/AbstractGeometry.php:
```diff
@@ -25,117 +26,141 @@
-    public function getSrid()
-    {
-        return $this->srid;
+        return json_encode($json);
```
`return json_encode($this);` same output because class implements `JsonSerializable`

Alexandre-T replied:
I added JsonSerializable because I read #194 , but (I'm not sure) I encountered an error in a specific case. Do you agree to wait this PR to be merged, then I will tried it.

tugrul replied:
@Alexandre-T `JsonSerializable` is good. I just mention that no need to create array to generate json from `toJson` function.

following function will be enough because `JsonSerializable` feature in use.

```php
public function toJson()
{
    return json_encode($this);
}
```

Alexandre-T commented:
I do agree with your four steps. What do you think about creating a 1.3.x-dev tag instead of a 1.x-dev?  

Changelog is already done. I could update the readme to add a small paragraph to avoid panic for developers using dev-master.

PowerKiKi commented on README.md:
```diff
@@ -1,26 +1,63 @@
+### MariaDB 10
+This version is **NOT** compatible with MariaDB version. Some spatial functions seems to work but their results are 
+different from MySQL version (StContains function is a good example). You can contribute, but I suggest to avoid 
+MySql and MariaDb servers, because of [their shortcomings and vulnerabilities](https://sqlpro.developpez.com/tutoriel/dangers-mysql-mariadb/).
```
The link to the article should be removed. Because it is a subjective opinion about RDBMS choice (expressed in a rather aggressive tone) and has no reason to be mentioned here. It also happens to be in French which most readers might not be able to read. And finally the way it is introduced make it looks like it is about MariaDB only, but it actually is about both MySQL and MariaDB.

As a matter of fact, the whole "Compatibility" section here is a bit confusing. Topics are mixed between "Compatibility between this lib and a RDBMS" and "Compatibility across RDBMS". The fact that `StContains` returns different result between MySQL and MariaDB does not make this lib incompatible with MariaDB.

rogeriolino commented:
Any news about this fix or some fork that fix it? I'm getting the same error after upgrating to PHP 8.

rogeriolino commented:
nevermind, I see that @Alexandre-T created a new one. Thank you!

PowerKiKi commented:
For the records, @rogeriolino  was referring to the fork of @Alexandre-T that is available there: https://github.com/Alexandre-T/doctrine2-spatial

Alexandre-T commented:
I created an organization named [LongitudeOne](https://github.com/longitude-one) and a new package that will replace and maintain this awesome spatial extension. All improvements from this PR are already included. I published a pre-released which is compatible with PHP8, MySQL5.7, MySQL8, PostgreSQL13 and PostGis 3.1.

sadortun commented:
@Alexandre-T we have a lot of people using this extension, if you want to volunteer as a maintainer, we  can simply add you here so everybody can use your updates

sadortun commented:
@Alexandre-T 

I just realize you've been working on this for the past year and I really appreciate your contributions.

Let me do what I mentioned https://github.com/creof/doctrine2-spatial/pull/209#issuecomment-653941286

And I'll be ready to merge this PR as. `2.x-beta`

samwilson commented:
Is there an idea of when a new version of this package might be released. I've been (happily) using `alexandret/doctrine2-spatial`, but it's now marked as abandoned.

Thanks for working on all this! It's super useful.

Alexandre-T commented:
I marked it abandoned, because I created this one: https://github.com/longitude-one/doctrine-spatial
Compatible with PHP8, all versions of Postgis, PostgreSQL and MySQL.
It is fully documented.
